src/quacc/settings.py

```diff
@@ -569,3 +570,33 @@ def change_settings(changes: dict[str, Any]):
     finally:
         for attr, original_value in original_values.items():
             setattr(SETTINGS, attr, original_value)
+
+
+def change_settings_wrap(func: Callable, changes: dict[str, Any]) -> Callable:
+    """
+    Wraps a function with the change_settings context manager if not already wrapped.
+
+    Parameters
+    ----------
+    func
+        The function to wrap.
+    changes
+        The settings to apply within the context manager.
+
+    Returns
+    -------
+    Callable
+        The wrapped function.
+    """
+    from quacc import change_settings
+
+    original_func = func.original_func if getattr(func, "__changed__", False) else func
+
+    @wraps(original_func)
+    def wrapper(*args, **kwargs):
+        with change_settings(changes):
+            return original_func(*args, **kwargs)
+
+    wrapper.__changed__ = True
+    wrapper.original_func = original_func
```

Comment:
Could we call this `._original_func` since it's not meant to be part of the public API? Thanks!

Reply:
Of course, thanks for suggestion. I'll update it.